build_tools/tests/promote_from_rc_to_final_test.py

```diff
@@ -0,0 +1,330 @@
+        + (" SUCCESS" if success else " FAILURE")
+    )
+    print(
+        "================================================================================="
+    )
+    return success
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description="""Tests promotion of packages from release candidate to final release (e.g. 7.10.0rc1 --> 7.10.0).
+"""
+    )
+    parser.add_argument(
+        "--platform",
+        help="OS platform: either 'linux' (default) or 'win'",
+        default="linux",
+    )
+    p = parser.parse_args(sys.argv[1:])
+    platform = p.platform
+    # make tmpdir
+    with tempfile.TemporaryDirectory(
+        prefix=f"PromoteRcToFinalTest-{platform}-"
+    ) as tmp_dir:
+        if platform == "linux":
+            # download some version
+            URL = "https://rocm.prereleases.amd.com/whl/gfx94X-dcgpu/"
+            version = Version("7.9.0rc1")
+            expected_version = Version("7.9.0")
+            packages = [
+                "rocm-7.9.0rc1.tar.gz",
+                "rocm_sdk_core-7.9.0rc1-py3-none-linux_x86_64.whl",
+                "rocm_sdk_devel-7.9.0rc1-py3-none-linux_x86_64.whl",
+                "rocm_sdk_libraries_gfx94x_dcgpu-7.9.0rc1-py3-none-linux_x86_64.whl",
+                "pytorch_triton_rocm-3.3.1+rocm7.9.0rc1-cp312-cp312-linux_x86_64.whl",
+                "torch-2.7.1+rocm7.9.0rc1-cp312-cp312-linux_x86_64.whl",
+                "torchaudio-2.7.1a0+rocm7.9.0rc1-cp312-cp312-linux_x86_64.whl",
+                "torchvision-0.22.1+rocm7.9.0rc1-cp312-cp312-linux_x86_64.whl",
+            ]
+        else:  # win
+            URL = "https://rocm.prereleases.amd.com/whl/gfx1151/"
+            version = Version("7.9.0rc1")
+            expected_version = Version("7.9.0")
+            packages = [
+                "rocm-7.9.0rc1.tar.gz",
+                "rocm_sdk_core-7.9.0rc1-py3-none-win_amd64.whl",
+                "rocm_sdk_devel-7.9.0rc1-py3-none-win_amd64.whl",
+                "rocm_sdk_libraries_gfx1151-7.9.0rc1-py3-none-win_amd64.whl",
+                "torch-2.9.0+rocm7.9.0rc1-cp312-cp312-win_amd64.whl",
+                "torchaudio-2.9.0+rocm7.9.0rc1-cp312-cp312-win_amd64.whl",
+                "torchvision-0.24.0+rocm7.9.0rc1-cp312-cp312-win_amd64.whl",
+                "pytorch_triton_rocm-3.3.1+rocm7.9.0rc1-cp312-cp312-linux_x86_64.whl",
+            ]
+
+        print(
+            f"Testing promotion of {version} to {expected_version} on platform {platform}"
+        )
+        print(f"  Downloading packages from {URL}")
+        for package in packages:
+            print(f"  Downloading {package}")
+            # otherwise it gets unhappy with the "+" in the URL
+            url_safe_encoding = URL + urllib.parse.quote(package)
+            print(url_safe_encoding)
+            subprocess.run(
+                ["curl", "--output", tmp_dir + "/" + package, url_safe_encoding],
+                check=True,
+            )
```

Comment:
Would be nice to cache these input files for iterating on the script/test. Pytest may have a plugin for that, or we could build something custom. Maybe optionally write to `$HOME/.therock/cache/[test name]` and then copy out of there into the temp dir to run the test